From e44de3751bdb193f20cd0b5636a09d5ffd02f9bd Mon Sep 17 00:00:00 2001
From: Don Olmstead <don.j.olmstead@gmail.com>
Date: Thu, 13 Jun 2024 16:13:08 -0700
Subject: [PATCH] Update search path for highway version

In version 1.2.0 of highway the defines used to determine the version moved from `highway.h` to `base.h`. See if the version is in `base.h` and if not look in `highway.h`.
---
 cmake/FindHWY.cmake | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/cmake/FindHWY.cmake b/cmake/FindHWY.cmake
index c5a90fbc..917f2e73 100644
--- a/cmake/FindHWY.cmake
+++ b/cmake/FindHWY.cmake
@@ -20,9 +20,19 @@ find_library(HWY_LIBRARY
 )
 
 if (HWY_INCLUDE_DIR AND NOT HWY_VERSION)
-  if (EXISTS "${HWY_INCLUDE_DIR}/hwy/highway.h")
-    file(READ "${HWY_INCLUDE_DIR}/hwy/highway.h" HWY_VERSION_CONTENT)
+  # Version 1.2.0 moved versioning to base.h
+  if (EXISTS "${HWY_INCLUDE_DIR}/hwy/base.h")
+    file(READ "${HWY_INCLUDE_DIR}/hwy/base.h" HWY_VERSION_CONTENT)
+    string(FIND "${HWY_VERSION_CONTENT}" "#define HWY_MAJOR" _FOUND_VERSION)
+    if (_FOUND_VERSION EQUAL -1)
+      unset(HWY_VERSION_CONTENT)
+      if (EXISTS "${HWY_INCLUDE_DIR}/hwy/highway.h")
+        file(READ "${HWY_INCLUDE_DIR}/hwy/highway.h" HWY_VERSION_CONTENT)
+      endif ()
+    endif ()
+  endif ()
 
+  if (HWY_VERSION_CONTENT)
     string(REGEX MATCH "#define HWY_MAJOR +([0-9]+)" _sink "${HWY_VERSION_CONTENT}")
     set(HWY_VERSION_MAJOR "${CMAKE_MATCH_1}")
 
-- 
2.45.2.windows.1
